brave/brave-core
# AC contribution change

Resolves https://github.com/brave/brave-browser/issues/6228

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
Plan 1
- start with short contribution interval and staging
- claim grant
- add one verified publisher to monthly tip and set AC to 50 BAT
- restart browser
- you shouldn't receive notification about not enough funds

Plan 2
- start with short contribution interval and staging
- claim grant
- add multiple verified publishers to monthly tip so that you are over balance
- restart browser
- you should receive notification about not enough funds

Plan 3
- start with short contribution interval and staging
- claim grant
- add one verified publisher and multiple unverified publishers to that total is over balance
- restart browser
- you shouldn't receive notification about not enough funds

Plan 4
- start with short contribution interval and staging
- claim grant (30 BAT)
- set AC budget to 50 BAT
- add verified publisher to AC table
- add verified monthly tips publishers so that total is 20 BAT
- wait for contribution to happen
- monthly tips and AC should go through, where AC amount would be 10

Plan 4
- start with short contribution interval and staging
- claim grant (30 BAT)
- set AC budget to 20 BAT
- populate AC table with verified and unverified publishers
- wait for contribution to happen
- make sure that 20 BAT went through and that only verified publishers received BAT

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
